src/main/java/edu/wpi/grip/ui/SocketControlView.java

```diff
@@ -0,0 +1,172 @@
+package edu.wpi.grip.ui;
+
+import com.google.common.base.Optional;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.Socket;
+import edu.wpi.grip.core.SocketHint;
+import edu.wpi.grip.core.events.ConnectionAddedEvent;
+import edu.wpi.grip.core.events.ConnectionRemovedEvent;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.control.*;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import org.controlsfx.control.RangeSlider;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ResourceBundle;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A JavaFX control that renders a {@link Socket}.  This control shows a label with the socket's identifier, as well as
+ * an optional control such as a slider or a number spinner that lets the user interact with the value in the socket.
+ * <p>
+ * This class both subscribes to and sends {@link SocketChangedEvent}s on the event bus, and acts as an intermediary
+ * between EventBus events and a JavaFX property.
+ */
+public class SocketControlView extends AnchorPane implements Initializable {
+    @FXML
+    private Label identifier;
+
+    @FXML
+    private StackPane controlPane;
+
+    private final EventBus eventBus;
+    private final Socket socket;
+    private final Property valueProperty;
+
+    public SocketControlView(EventBus eventBus, Socket<?> socket) {
+        checkNotNull(eventBus);
+        checkNotNull(socket);
+
+        this.eventBus = eventBus;
+        this.socket = socket;
+        this.valueProperty = new SimpleObjectProperty();
+
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("SocketControl.fxml"));
+            fxmlLoader.setRoot(this);
+            fxmlLoader.setController(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Actually, the real reason that I want to have a base uncachable exception is so that we can put methods inside it like `toMarkdown` or `toIssue` so that we can show a UI-screen to allow users to copy the error into an issue really easily.